janusgraph-hbase-parent/janusgraph-hbase-core/src/main/java/org/janusgraph/diskstorage/hbase/HBaseStoreManager.java

```diff
@@ -891,21 +896,45 @@ private void setCFOptions(HColumnDescriptor cdesc, int ttlInSeconds) {
                     }
 
                     for (StaticBuffer b : mutation.getDeletions()) {
+                        // commands.getSecond() is a Delete for this rowkey.
                         commands.getSecond().deleteColumns(cfName, b.as(StaticBuffer.ARRAY_FACTORY), delTimestamp);
                     }
                 }
 
                 if (mutation.hasAdditions()) {
-                    if (commands.getFirst() == null) {
-                        Put p = new Put(key, putTimestamp);
-                        commands.setFirst(p);
-                    }
-
+                    // All the entries (column cells) with the rowkey use this one Put, except the ones with TTL.
+                    Put p = new Put(key, putTimestamp);
+                    
                     for (Entry e : mutation.getAdditions()) {
```

Comment:
add a comment here: "At the end of this loop, there will be one Put entry in the commands.getFirst() list that contains all additions without TTL set, and one Put entry for each column that has TTL set."